src/include/processor/processor_task.h

```diff
@@ -14,14 +14,18 @@ class ProcessorTask : public common::Task {
 
     void run() override;
     void finalizeIfNecessary() override;
+    void addSource(PhysicalOperator* source);
 
 private:
     static std::unique_ptr<ResultSet> populateResultSet(
         Sink* op, storage::MemoryManager* memoryManager);
 
+    void updateProgress() const override;
+
 private:
     bool sharedStateInitialized;
     Sink* sink;
+    std::vector<PhysicalOperator*> sourceOps;
```

Comment:
By definition, a process task can only have 1 source. So just use `PhysicalOperator*`

Reply:
Same as the reply to the comment below